app/src/main/java/com/example/reflector_android/network/XmlParser.kt

```diff
@@ -0,0 +1,135 @@
+package com.example.reflector_android
+
+import android.util.Xml
+import com.example.reflector_android.network.Article
+import okio.IOException
+import org.xmlpull.v1.XmlPullParser
+import org.xmlpull.v1.XmlPullParserException
+import java.io.InputStream
+import java.lang.IllegalStateException
+
+
+class XmlParser {
+    private val ns: String? = null
+
+    @Throws(XmlPullParserException::class, IOException::class)
+    suspend fun parse(inputStream: InputStream): MutableList<Article> {
+        inputStream.use { inputStream ->
+            val parser: XmlPullParser = Xml.newPullParser()
+            parser.setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, false)
+            parser.setInput(inputStream, null)
+            parser.nextTag()
+            return readFeed(parser)
+        }
+    }
+
+    suspend private fun readFeed(parser: XmlPullParser): MutableList<Article> {
+        val items = mutableListOf<Article>()
+
+        parser.require(XmlPullParser.START_TAG, ns, "rss")
```

Comment:
Move "rss" to a Constant. Typically in programming, you want to minimize stringly-typed stuff because you're more likely to misspell "rss" if you need to move stuff around since it's not autocompleted.

Also if further down the road our API decides to change the XML tag for "item" to "object" we only have to change your constant declaration and that's it.

At the top of your XmlParser class, do this:
```
class XmlParser {
   companion object Tags {
        val rss = "rss"
        val item = "item"
        var title = "title"
        ....
    }
}
```

Then anywhere you type strings for "rss" or "item" or whatever you can just do Tags.[tag_you_want].